bot/exts/help_channels/_cog.py

```diff
@@ -642,3 +644,16 @@ async def helpdm_command(
         else:
             await _caches.help_dm.delete(ctx.author.id)
         await ctx.send(f"{constants.Emojis.ok_hand} {ctx.author.mention} Help DMs {state_str}!")
+
+    @tasks.loop(hours=1)
+    async def check_channel_consistency(self) -> None:
```

Comment:
I believe there is a possible race condition which would extend the amount of bugs.

If a channel is currently being closed while this runs, they would be awaiting an edit coroutine. This would allow the program to switch context and run the waiting task here.

However, the api would still be completing when this task decides to reload the cog, furthering the issue. 

Its a race condition that would make the problem worse, imo. 

A solution with that could be some asyncio.Locks, in order to not let the context switch to within here before a channel finishes closing and returns.

Reply:
I've looked into it when we opened the issue, but the cog is pretty resilient. If the closing process is still on the bot side, that shouldn't affect matters since it'll be canceled and resynced on startup.

If it's waiting on an API call from discord, I think it actually does not cancel until it's done, but I may be mistaken. Either way, the worst case scenario if it doesn't is that we'll move one channel twice, but it'll be to the same place both times, so nothing will break.